addons/stock/models/stock_move_line.py

```diff
@@ -241,36 +241,27 @@ def write(self, vals):
         moves_to_recompute_state = self.env['stock.move']
         Quant = self.env['stock.quant']
         precision = self.env['decimal.precision'].precision_get('Product Unit of Measure')
-        # We forbid to change the reserved quantity in the interace, but it is needed in the
-        # case of stock.move's split.
-        # TODO Move me in the update
-        if 'product_uom_qty' in vals:
-            for ml in self.filtered(lambda m: m.state in ('partially_available', 'assigned') and m.product_id.type == 'product'):
-                if not ml._should_bypass_reservation(ml.location_id):
-                    qty_to_decrease = ml.product_qty - ml.product_uom_id._compute_quantity(vals['product_uom_qty'], ml.product_id.uom_id, rounding_method='HALF-UP')
-                    try:
-                        Quant._update_reserved_quantity(ml.product_id, ml.location_id, -qty_to_decrease, lot_id=ml.lot_id, package_id=ml.package_id, owner_id=ml.owner_id, strict=True)
-                    except UserError:
-                        if ml.lot_id:
-                            Quant._update_reserved_quantity(ml.product_id, ml.location_id, -qty_to_decrease, lot_id=False, package_id=ml.package_id, owner_id=ml.owner_id, strict=True)
-                        else:
-                            raise
-
         triggers = [
             ('location_id', 'stock.location'),
             ('location_dest_id', 'stock.location'),
             ('lot_id', 'stock.production.lot'),
             ('package_id', 'stock.quant.package'),
-            ('result_package_id', 'stock.quant.package'),
             ('owner_id', 'res.partner')
         ]
         updates = {}
         for key, model in triggers:
             if key in vals:
                 updates[key] = self.env[model].browse(vals[key])
 
-        if updates:
+        if updates or 'product_uom_qty' in vals:
```

Comment:
It's kind of weird to have this `updates` dict which just includes the browse records of the `vals` relational fields. But ok, I guess this would be more refactoring than fixing to change that.

Reply:
the `_update_reserved_quantity` signature of `stock.quant` receives recordset, not record ids:

``` python
_update_reserved_quantity(self, product_id, location_id, quantity, lot_id=None, package_id=None, owner_id=None, strict=False):
```

the purpose of this dict was to be able to write the more succinct
``` python
location_id=updates.get('location_id', ml.location_id),
```
instead of 
``` python
location_id=vals.get('location_id') and self.env['stock.location'].browse(vals['location_id']) or ml.location_id),
```
when using this method. But it doesn't help the reader indeed.

Reply:
Sure I see the point :)